cpp/src/arrow/io/file.cc

```diff
@@ -246,31 +246,55 @@ static inline Status FileRead(int fd, uint8_t* buffer, int64_t nbytes,
   }
   *bytes_read = static_cast<int64_t>(_read(fd, buffer, static_cast<uint32_t>(nbytes)));
 #else
-  *bytes_read = static_cast<int64_t>(read(fd, buffer, static_cast<size_t>(nbytes)));
+  int64_t n = 0;
+  int64_t N = nbytes;
```

Comment:
Since `N` is never changed, just use `nbytes`? Maybe make it `const int64_t nbytes` in the function arguments for good measure